tools/fuzzer/ast.h

```diff
@@ -268,6 +383,23 @@ class TernaryExpr {
   std::unique_ptr<Expr> rhs_;
 };
 
+class CastExpr {
+ public:
+  static constexpr int PRECEDENCE = 2;
+
+  CastExpr(Type type, Expr expr);
+
+  const Type& type() const;
+  const Expr& expr() const;
+  int precedence() const { return PRECEDENCE; }
+
+  friend std::ostream& operator<<(std::ostream& os, const CastExpr& expr);
+
+ private:
+  Type type_;
+  std::unique_ptr<Expr> expr_;
```

Comment:
Why `expr_` is a pointer, but `type_` is not?

Reply:
Because `Expr` already contains `CastExpr` as a possible variant type, hence a definition of `Expr expr_` would mean that `CastExpr` is of infinite size.